undertow/WFLY-12459_Active_Request_Tracking.adoc

```diff
@@ -0,0 +1,143 @@
+= WFLY-12473 Active Request Tracking
+:author:            Flavia Rainone
+:email:             frainone@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-12459
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/UNDERTOW-1593
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+Undertow, WildFly Undertow Subsystem
+
+=== Other Interested Projects
+
+== Requirements
+The new feature consists of adding a way for users to retrieve active requests, ie.
+requests that haven't finished and are active on the server.
+There was a similar feature in EAP6 and the plan is to be able to provide something similar
+in EAP7 with Undertow.
+
+Since there is an added impact for tracking active requests, the feature has to be
+enabled to be used. To enable the feature, we are adding an attribute to Undertow
+subsystem, called `"active-request-statistics-enabled"`.
+
+That attribute can only be set to `true` if `"statistics-enabled"` is `true`.
+
+Once that feature is enabled, there is an added resource to undertow server
+called "active-request-tracker=enabled"
+As an example, look at the cli script below:
+
+....
+subsystem=undertow/set-attribute(name="active-request-statistics-enabled", "true")
+....
+
+As a result, we will have a new resource in the Undertow Subsystem with the name:
+
+.....
+subsystem=undertow/active-request-tracker=enabled
+.....
+
+This tracker can later be referenced for reading the active requests with the operations:
+....
+list-active-requests
+list-active-requests(format="same as an Undertow access log formatting")
+list-active-requests-with-id
+list-active-requests-full
+active-request-info(id="id")
+....
+See the example bellow:
+----
+subsystem=undertow/active-request-tracker=requestTracker:list-active-requests
+----
+
+The operation below will show just the requested attributes from the request:
+----
+subsystem=undertow/active-request-tracker=requestTracker:list-active-requests(attributes="bytes-sent,bytes-received")
+----
+
+Notice that, in order for the track to start tracking actual requests, it must be referenced as a filter inside the server as in:
+----
+/subsystem=undertow/server=default-server/host=default-host/filter-ref=requestTracker:add
+----
```

Comment:
If I understand this, then this filter reference will be kind of a reference to a `virtual` filter, right? How is this reflected in the configuration XML? What if user defines its own filter named `requestTracker`? Shall we deny this name?

Reply:
AIUI the intent in the above bit is to describe the automatic internal behavior as being the same as what would happen if a filter-ref was actually in the config. Not that there would actually be such a thing in the config, and thus no "add" operation to set it up.